nuxeo-features/rest-api/nuxeo-rest-api-server/src/main/java/org/nuxeo/ecm/restapi/server/jaxrs/OAuth2Object.java

```diff
@@ -188,54 +192,194 @@ public Response getToken(@PathParam("providerId") String providerId, @Context Ht
     @GET
     @Path("token")
     public List<NuxeoOAuth2Token> getTokens(@Context HttpServletRequest request) {
-        checkPermission();
+        checkPermission(null);
         return getTokens();
     }
 
+    /**
+     * Retrieves an OAuth2 provider token.
+     *
+     * @since 10.2
+     */
+    @GET
+    @Path("token/provider/{providerId}/user/{nxuser}")
+    public Response getProviderToken(@PathParam("providerId") String providerId, @PathParam("nxuser") String nxuser,
+                                     @Context HttpServletRequest request) {
+        checkPermission(nxuser);
+        NuxeoOAuth2ServiceProvider provider = getProvider(providerId);
+        return Response.ok(getToken(provider, nxuser)).build();
+    }
+
     /**
      * Retrieves an OAuth2 Token.
      *
      * @since 9.2
+     *
+     * @deprecated since 10.2 Use getProviderToken instead.
```

Comment:
Can you replace `getProviderToken` by `{@link #getProviderToken(String, String, HttpServletRequest)}` in order to have javadoc bindings.